components/server/ee/src/prebuilds/github-service.ts

```diff
@@ -26,7 +26,8 @@ export class GitHubService extends RepositoryService {
     @inject(GithubContextParser) protected githubContextParser: GithubContextParser;
 
     async getRepositoriesForAutomatedPrebuilds(user: User): Promise<ProviderRepository[]> {
-        const repositories = (await this.githubApi.run(user, (gh) => gh.repos.listForAuthenticatedUser({}))).data;
+        const octokit = await this.githubApi.create(user);
+        const repositories = await octokit.paginate(octokit.repos.listForAuthenticatedUser, { per_page: 100 });
```

Comment:
Cool to see there is a library method for this! :+1:

We could try to push our `map` function into `paginate` to be more conscious about memory usage: https://github.com/octokit/plugin-paginate-rest.js/#octokitpaginate.
Because we do not have a `filterMap`, it could return `ProviderRepository | undefined`, followed by a `filter(repo => !!)` afterwards.

The reason I'm raising this seemingly minor issue is that when looking at COU profiles, our code is riddled with uncertainties about cardinality, leading to performance drops/spikes.
Also happy to review a follow up for this.

Reply:
Good point! I believe we can simply move our after-the-fact `filter` into the `octokit.paginate` call (not sure what you mean with a `filterMap`, but I think a simple `filter` might do the trick).

Reply:
Ah, whoops, the `map` would indeed have a much stronger memory reduction effect than the `filter` alone. 💡 Follow-up issue to the rescue!